### PR TITLE
DM-9624: AP Science Lead -- Bring Alg Components up to date

### PR DESCRIPTION
**Pull request by ebellm**
Reduce algorithmic detail specified in:

   PSF Estimation - Single CCD PSF Estimation
   Astrometric Fitting - Single CCD
   PSF Matching -Image Subtraction
   Association and Matching - Single CCD to Reference Catalog, Semi-Blind
   Association and Matching - Single Visit to Reference Catalog, Semi-Blind

Respond to RHL comments throughout these sections.

Also various typo and reference fixes.

### CODE REVIEW

File: DM_Applications_Design.tex [hunk 1]
```diff
@@ -139,7 +139,7 @@
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%% Command's for merging Robert's document, at least temporarily.
-%\usepackage[top=1cm, bottom=2cm, left=2cm, right=2cm]{geometry}
```
Comment (SimonKrughoff) [inline DM_Applications_Design.tex]: It's not super important, but we usually try to keep whitespace changes in a separate commit unrelated to other changes.  This is more important in code, but I thought I'd mention it.

File: sections/algorithmic_components.tex [hunk 2]
```diff
@@ -701,11 +697,10 @@ \subsection{Astrometric Fitting}
...
 
-We can project the per-chip coordinates into a flat (Gnomonic) projection in the focalplane.  This will take out the bulk of the optical distortion.  Next, we will use a matching algorithm like that outlined in \cite{tabur04}.  Once we match, a 2D polynomial solution will be fit to minimize the residual.
+We can project the per-chip coordinates into a flat projection in the focalplane.  This will take out the bulk of the optical distortion.  We will then match the science and template sources and fit a distortion model to minimize the residual positional offsets.
```
Comment (SimonKrughoff) [inline sections/algorithmic_components.tex]: It actually looks like we will not match in focalplane coordinates but on the surface of a unit sphere.  This has some nice properties and serves the same purpose of taking out the bulk optical distortion.  I say this based on the recent work by Chris M. (DMTN forthcoming).  I don't see a reason not to state this as the baseline right now.

File: sections/algorithmic_components.tex [hunk 3]
```diff
@@ -715,7 +710,7 @@ \subsubsection{Single Visit}
 \label{sec:acSingleVisitAstrometricFit}
 Full visit astrometric fitting will be done as a bootstrapping step toward higher quality calibration in DRP.  All measurements in the visit will be projected to a tangent plane, taking into account all knowledge of the sensor arrangement and optics.  The reference catalog (likely the DRP reference catalog) will be projected to the same tangent plane.
 
-Sources will be matched, again using a \cite{tabor04} like algorithm.  Once the reference and observations are matched, a multi-component WCS will be fit.  We expect the components will be related to residuals on the optical model and will included a component to account for the Von Karman turbulence.
+Sources will be matched, again using a \cite{tabur2007}-like algorithm.  Once the reference and observations are matched, a multi-component WCS will be fit.  We expect the components will be related to residuals on the optical model and will included a component to account for the Von Karman turbulence.
```
Comment (SimonKrughoff) [inline sections/algorithmic_components.tex]: Is it obvious that when we say "Von Karman turbulence" that we simply mean the atmospheric component?

File: sections/algorithmic_components.tex [hunk 4]
```diff
@@ -795,41 +790,59 @@ \subsection{Retrieve Diffim Template for a Visit}
...
+functions $\kappa(u,v) = \sum_i a_i \kappa_i(u,v)$, where the coefficients
+are determined via ordinary least-squares estimation.  The basis functions
+$\kappa_i(u,v)$ are a degree of freedom in this problem.
+\footnote{The current implementation of the A\&L matching algorithm is
+summarized in detail by Becker et al. (2013) (\url{http://ls.st/x9f}).}
+
+A spatially-invariant matching kernel $\kappa(u,v)$ is determined separately for image substamps centered on multiple kernel candidates across the image after brighter-fatter corrections have been applied.  The kernel candidates are selected using the {\tt DiaCatalogSourceSelector} to query the appropriate reference catalog for sources to use for PSF matching. This selector allows the user to specify the brightness and color range of the objects, toggle star or galaxy selection, and to include variable objects or not. Sources are vetted for signal-to-noise and masked pixels (in both the template and science image). The matching (spatially-invariant) kernel models $\kappa_j(u,v)$, determined for each kernel candidate $j$ as described above, are examined and filtered by various quality measures.
```
Comment (SimonKrughoff) [inline sections/algorithmic_components.tex]: I'm a little uncomfortable talking about specific class names in this document.  I think it indicates we are too close to the implementation and I'm afraid it will make it hard to maintain.  I think just generalizing to say "a star selector" or something like that.

File: sections/algorithmic_components.tex [hunk 5]
```diff
@@ -1155,64 +1168,27 @@ \subsection{Association and Matching}
...
-sensor and reference catalog.  
+Given a set of sources detected on a single sensor, and a corresponding reference catalog, we adopt a simple pattern matching algorithm to cross-match between the catalogs.  We assume that the sources detected on the sensor have approximate positions given by the telescope's initial estimate of a WCS, that we know the plate scale of the system, and that positional errors are available for both the sensor and reference catalog.
 
-Cross-matching is undertaken using the Optimistic Pattern Matching (B) algorithm of Tabur \cite{tabur2007}. The algorithm defines an order $m$ (default 6) size $m-1$ acyclic connected tree as the pattern to match between catalogs. The details of the LSST implementation of the Optimistic Pattern Matching (B) algorithm are as follows:
+Cross-matching may be undertaken using the Optimistic Pattern Matching (B) algorithm of Tabur \cite{tabur2007}. The algorithm defines an order $m$ (default 6) size $m-1$ acyclic connected tree as the pattern to match between catalogs.  Trees are constructed and matched for the brightest $n$ stars (default = 50) in descending order until a match is found.
```
Comment (SimonKrughoff) [inline sections/algorithmic_components.tex]: Again, just because we are working on this now, things are changing a bit and we might as well take them into account.  I'm not sure we should mention defaults here.  Since they are really implementation details.  If you do want to include them, I think the number of bright stars will be closer to 200 than 50.  I'm not sure what the default number of legs on the acyclic graph is, but we can check.

File: sections/algorithmic_components.tex [hunk 6]
```diff
@@ -1155,64 +1168,27 @@ \subsection{Association and Matching}
...
-  brightest object from $I$ and finds the next brightest $m$ sources.
-\end{itemize} 
-
-Once $m-1$ edges match, an initial verification step is performed where a gnomonic projection of the reference catalog is fit to the matched stars. Given a consistent fit to the gnomonic projection all sources and reference objects are matched in sensor coordinates (sorted by brightness) with the brightest source in $I$ selected when two sources are within the match tolerance (default 3 arcsec). 
+After a matched solution is found, we will verify the quality of the match by comparing the positions of the science and reference objects.
 
 For the case of no WCS for the sensor or a significant error in the
-WCS ($> 3$ arcsec), a blind matching will be undertaken using the
-algorithms in astrometry.net. 
+WCS ($> 3$ arcsec), blind matching will be undertaken (e.g., by using the
```
Comment (SimonKrughoff) [inline sections/algorithmic_components.tex]: I wonder if we should just take this out.  Right now Chris's algorithm is matching for offsets up to 30 arcsec and I don't know if we want a dependence on a.net in the design document.  If you think we need to hedge our bets on blind solving, we could just say "a blind solver" rather than specifying a.net specifically.

**Review by SimonKrughoff — APPROVED** [review]
Looks great overall.

**Comment by timj** [comment]
Would anyone object if I rebased this PR? The changes I made for #37 caused trouble with this PR and I feel like I should fix it.

File: sections/algorithmic_components.tex [hunk 7]
```diff
@@ -795,41 +790,59 @@ \subsection{Retrieve Diffim Template for a Visit}
...
+the image; during convolution and correlation there is an implicit
+summation over $u,v$.} Then the difference image, upon which new or
+variable sources are detected, is given by $D = S - (\kappa \otimes
+T)$.
+We model the PSF--matching kernel by decomposing it into a set of basis
+functions $\kappa(u,v) = \sum_i a_i \kappa_i(u,v)$, where the coefficients
+are determined via ordinary least-squares estimation.  The basis functions
+$\kappa_i(u,v)$ are a degree of freedom in this problem.
+\footnote{The current implementation of the A\&L matching algorithm is
+summarized in detail by Becker et al. (2013) (\url{http://ls.st/x9f}).}
```
Comment (timj) [inline sections/algorithmic_components.tex]: This should be a proper reference to the DocuShare document. I note that in this case the URL actually points to a trac attachment rather than the real DocuShare location which should be LDM-227.

**Comment by ebellm** [comment]
Sure, @timj, you can rebase.

**Comment by timj** [comment]
Ok. Rebased. Changes look a bit clearer now. There is now inconsistent use of newlines in paragraphs now but I don't think we can do much about that given the history of the document. These days I tend to use one line per sentence.

**Comment by timj** [comment]
The normal policy for DM *code* development is to rebase ticket branches rather than continually confuse the history by doing incremental merges from the other branch. I assume we prefer that approach for document development in Git as well. If you were rebasing there would be no need for the conflict resolution merge commit since the conflict would be resolved as part of the rebase (and I see this branch has just been merged).

**Comment by ebellm** [comment]
Apologies, @timj.  I was attempting to follow the developer guide but can see now where I erred.
